### PR TITLE
Spec: Documented the branching strategy

### LINKED ISSUE #65: Document branching strategy
## ⚠️ Problem
Everyone is currently working out of main but that will eventually cause issues when features have interim PRs that are not ready to roll out yet.

## 🛠️ Solution
Document a branching strategy where devs work out of their own branches/forks, then commit back to a feature branch, which then gets merged into a **dev** branch. When the release is ready, we'll merge **dev** to **main**.

## ℹ️ Additional context
This should be done before closing any PRs that will not be included in the first release.

## 🙋‍♀️ Ask for the community

We could use your help:
1. Please vote this issue up (👍) to prioritize it.
2. Leave comments if you'd like to share any thoughts to improve the plans.

### PR DESCRIPTION
**Pull request by flanakin**
## 🛠️ Description
Outlines a branching strategy with main (stable), dev (next release), feature branches, and personal branches.

Fixes #65

### CODE REVIEW

File: src/branching.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Branching strategy
+
+- [main](https://github.com/microsoft/cloud-hubs/tree/main) includes the latest stable release.
+- [dev](https://github.com/microsoft/cloud-hubs/tree/dev) includes the latest changes that will go into the next release.
+- Feature branches (`features/<feature-name>`) are used for any in-progress features that are not yet ready for release.
+- Personal branches (`<your-github-account>/<feature-name>`) are intended for a single developer and typically not shared. Use these for small changes that can easily be integrated into the next release.
+
+```mermaid
```
Comment (flanakin) [inline src/branching.md:8]: Use the "rich diff" to see what this looks like:
image

**Review by MSBrett — APPROVED** [review]
Approved

**Review by ankur-ms — APPROVED** [review]

File: src/README.md [hunk 2]
```diff
@@ -120,24 +149,25 @@ Every PR is expected to include some sort of verification:
...
-9. INTERNAL ONLY: When your PR is ready to be reviewed, add the **Needs: Review** label. We only review PRs with this label. We strive to review these PRs at least once per business day (Pacific time, excluding US holidays). We will remove the label after reviewed.
-10. As you update your PR and apply changes, mark each conversation as [resolved](https://docs.github.com/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#resolving-conversations). After all comments are addressed, add the **Needs: Review** label again to signify that it is ready to review.
-11. If you run into any merge issues, checkout this [Git tutorial](https://github.com/skills/resolve-merge-conflicts) to help you resolve merge conflicts and other issues.
+4. Only address 1 issue per pull request and [link the issue in the pull request](https://github.com/blog/957-introducing-issue-mentions).
+5. Be sure to follow our [coding guidelines](./code.md) and keep code changes as small as possible.
+6. Validate changes by running locally and running [[unit tests]].
```
Comment (tsilvers-ms) [inline src/README.md:164]: Do we want to specify a test coverage requirement? Enforce via build process?

**Review by tsilvers-ms — APPROVED** [review]

File: src/README.md (hunk 2 above)
Comment (flanakin) [reply src/README.md:164]: I would love that, but I think we need to establish a system for unit testing first. I think we need to do some research to figure out how to accomplish that for templates. Let's document the requirement as part of establishing that unit testing process.

**Pull request merged**